lib/modem_info/modem_info_json.c

```diff
@@ -235,16 +243,8 @@ int modem_info_json_string_encode(struct modem_param_info *modem, char *buf)
 		}
 
 		total_len += ret;
-		json_add_obj(data_obj, "deviceInfo", device_obj);
-	}
-
-	if (total_len > 0) {
-		memcpy(buf,
-			cJSON_PrintUnformatted(data_obj),
-			MODEM_INFO_JSON_STRING_SIZE);
+		json_add_obj(root_obj, "deviceInfo", device_obj);
 	}
 
-	cJSON_Delete(data_obj);
```

Comment:
Is the cleanup done elsewhere now?

Reply:
Surprisingly, yes. The cloud transmission frees the object after it is complete. I'll add a comment.

Reply:
Yes, please add it, I was wondering as well who frees the `root_obj` until I saw this comment.